corehq/apps/geospatial/utils.py

```diff
@@ -31,16 +31,25 @@ def _format_coordinates(lat, lon):
     return f"{lat} {lon} 0.0 0.0"
 
 
-def set_case_gps_property(domain, case_data):
+def set_case_gps_property(domain, case_data, create_case=False):
     location_prop_name = get_geo_case_property(domain)
-    helper = CaseHelper(domain=domain, case_id=case_data['id'])
-    case_data = {
+    data = {
         'properties': {
             location_prop_name: _format_coordinates(case_data['lat'], case_data['lon'])
         }
     }
 
-    helper.update(case_data)
+    if create_case:
+        helper = CaseHelper(domain=domain)
+        data |= {
+            'case_name': case_data['name'],
+            'case_type': case_data['case_type'],
+            'owner_id': case_data['owner_id'],
+        }
+        helper.create_case(data)
+    else:
+        helper = CaseHelper(domain=domain, case_id=case_data['id'])
```

Comment:
nit: isn't it possible to have helper initiated on top with just domain and then add case_id for update later?

Reply:
We could do this, and then simply set `case_id` with `helper.case.case_id = case_data['id']`.

Reply:
Ah, I see `CaseHelper.case` only gets created if `case_id` is passed when initializing `CaseHelper`. To make this work we would have to extract the `BriefCase` dataclass out of `CaseHelper` and then do `helper.case = BriefCase(case_id=case_data['id], domain=domain)`. We could also initialize `CaseHelper` at the top with a meaningless `case_id` so that it sets the case field, but this feels less clean.

Given the above, I'll keep the current implementation and leave managing the `case` field to the `CaseHelper` class.